.github/workflows/build-pr.yaml

```diff
@@ -0,0 +1,64 @@
+name: builds on rudder-transformer prs
+on:
+  pull_request:
+    branches:
+      - feat-github-actions
+
+env:
+  DOCKER_USER: ${{secrets.DOCKER_USER}}
+  DOCKER_PASSWORD: ${{secrets.DOCKER_PASSWORD}}
+  REPO_NAME: "rudderlabs/develop-rudder-transformer"
+
+jobs:
+  build-transformer:
+    if: github.event_name == 'pull_request'
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [14.x]
+    steps:
+      -
+        name: Check out repository code
+        uses: actions/checkout@v3
+      -
+        name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: rudderstack/rudder-transformer
+      -
+        name: Extract Version
+        id: vars
+        shell: bash
+        run: |
+          echo "##[set-output name=version;]$(echo ${GITHUB_REF#refs/heads/})"
+      - 
+        name: Build the image for tests
+        run: docker build . --file Dockerfile --tag ${{ env.REPO_NAME }}:${{ steps.vars.outputs.version }}
+      -
+        name: Run npm tests - testDestinations on Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+        run: |
+          docker run${{ env.REPO_NAME }}:${{ steps.vars.outputs.version }} npm run testDestinations
+      -
+        name: Run npm tests - testVersionedRouter on Node.js ${{ matrix.node-version }}
+        run: |
+          docker run ${{ env.REPO_NAME }}:${{ steps.vars.outputs.version }} npm run testVersionedRouter
```

Comment:
if this step is successful, we can push the docker image directly. isn't building the image again in line 57 redundant ?